src/components/Navbar.jsx

```diff
@@ -0,0 +1,66 @@
+import { useState, useRef, useEffect } from 'react';
+import { useOnClickOutside } from "../useOnClickOutside";
+import '../style/navbar.css';
+import { MdClose } from 'react-icons/md';
+import { FiMenu } from 'react-icons/fi';
+
+
+const Navbar = () => {
+    const [dropdown, setDropdown] = useState(false);
+    const [navbarOpen, setNavbarOpen] = useState(false);
+    const ref = useRef();
+    useEffect(() => {
+      const handler = (event) => {
+        if (
+          navbarOpen &&
+          ref.current &&
+          !ref.current.contains(event.target)
+        ) {
+          setNavbarOpen(false);
+        }
+      };
```

Comment:
[OPTIONAL]
Check the number of spaces for every line to structure the code. sometimes you do 4 spaces, sometimes you do 2 of indexation.